src/common/types.ts

```diff
@@ -1,18 +1,40 @@
-import { asObject, asString } from 'cleaners'
+import { asObject, asString, Cleaner } from 'cleaners'
 import { EdgeToken, EdgeTokenInfo, EdgeTransaction } from 'edge-core-js/types'
 
 export const DATA_STORE_FILE = 'txEngineFolder/walletLocalData.json'
 export const TXID_MAP_FILE = 'txEngineFolder/txidMap.json'
 export const TXID_LIST_FILE = 'txEngineFolder/txidList.json'
 export const TRANSACTION_STORE_FILE = 'txEngineFolder/transactionList.json'
 
+// Same as asOptional but will not throw if cleaner fails but will
+// return the fallback instead
+export function asSafeOptional<T>(cleaner: Cleaner<T>, fallback: T): Cleaner<T>
+export function asSafeOptional<T>(cleaner: Cleaner<T>): Cleaner<T | undefined>
+export function asSafeOptional(cleaner: any, fallback?: any) {
+  return (raw: any) => {
+    if (raw != null) {
+      try {
+        return cleaner(raw)
+      } catch (e) {
+        // Do nothing. return fallback
+      }
+    }
+    return fallback
+  }
+}
```

Comment:
This is called `asMaybe`, and already includes the fallback behavior. You should use that instead.